constellation-app/constellation
# File Importer - Only allow importing of multiple files with same column structure

### Prerequisites


    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change
File Importer enforce all files that are being added have the same structure (so that the mapping makes sense across all files)


### Alternate Designs



### Why Should This Be In Core?
Better UX


### Benefits
Better UX


### Possible Drawbacks



### Verification Process

- Open File Importer
- Add 2 different files with different column structures (number of columns or the column headings) to the importer ( 'Files Include Headers' option is enabled by default)
- Repeat with different combinations of files.
- A warning message should be displayed and files with a different structure (to the 1st file added) should be removed.
- Disable 'Files Include Headers' option and repeat. This should only warn if the number of columns are different.



### Applicable Issues
https://github.com/constellation-app/constellation/issues/1102
